File: resource/java/com/template/Goods.java
package com.template;
import java.io.DataInputStream;
public class Goods implements TemplateInterface
{
	public String good_id;
	public String name;
	public short color;
	public short type;
	public int level;
	public String des;
	public String icon_id;
	public short can_use;
	public String use_effect;
	public void read(DataInputStream in)throws Exception
	{
		good_id=in.readUTF();
		name=in.readUTF();
		color=in.readShort();
		type=in.readShort();
		level=in.readInt();
		des=in.readUTF();
		icon_id=in.readUTF();
		can_use=in.readShort();
		use_effect=in.readUTF();
	}
}

File: resource/cpp/com/template/Goods.cpp
#include "Goods.h"
void Goods::read(ByteBuffer* buffer)
{
		this->good_id=buffer->readUTF();
		this->name=buffer->readUTF();
		this->color=buffer->readShort();
		this->type=buffer->readShort();
		this->level=buffer->readInt();
		this->des=buffer->readUTF();
		this->icon_id=buffer->readUTF();
		this->can_use=buffer->readShort();
		this->use_effect=buffer->readUTF();
}
